modules/doi/includes/utilities.inc

```diff
@@ -864,3 +833,38 @@ function islandora_doi_get_mods($id) {
 
   return islandora_doi_translate($source_string, $content_type);
 }
+
+/**
+ * Filter HTML tags from imported XML metadata.
+ *
+ * @param string $text
+ *   The text of the metadata to be filtered 
+ * @param array $removals
+ *   An array of strings to be removed from $text prior to filtering
+ *
+ * @return string 
+ *   String filtered using value of $islandora_doi_handle_improted_metadata 
+ */
+function islandora_doi_filter_html($text, $removals = FALSE) {
+  if ($removals) {
+    foreach ($removals as $removal) {
+      $text = str_replace($removal, '', $text);
+    }
+  }
+
+  switch (variable_get('islandora_doi_handle_imported_metadata', 'strip')) {
+    case 'strip':
+      $text = strip_tags($text);
+      break;
+
+    case 'encode':
+      $text = htmlspecialchars($text);
+      break;
+
+    default:
+      break;
```

Comment:
Should we perhaps issue a warning if we get here, as that should _never_ happen?

Reply:
Sure, why not.

Reply:
^ that sounds way more flippant than I intended. What I meant to say is:
Yes, this is a good idea and I see no reason not to do this!